### PR TITLE
Made some cmd pages

### PR DESCRIPTION
**Pull request by ahujatejas06**
Made some cmd pages

Logo needs to be edited... just added to check

### CODE REVIEW

File: docs/overview.md [hunk 1]
```diff
@@ -3,3 +3,21 @@ title : Overview
...
 ---
+# Overview
+The program is divided into multiple sections for the easy management of the different parts. The different sections of the program use separate ports for increased stability. A crash at one of the ports/handler leaves the others remain unaffected.
+
+- RAS: 3470
+-  AUTH: 3475
+-  STUDENT: 3480
+-  COMPANY: 3485
+-  ADMIN:
+-    RC: 3490
```
Comment (1-Harshit) [inline docs/overview.md:14]: Try nested list

File: docs/overview.md [hunk 2]
```diff
@@ -3,3 +3,21 @@ title : Overview
 layout: default
 nav_order: 2
 ---
+# Overview
+The program is divided into multiple sections for the easy management of the different parts. The different sections of the program use separate ports for increased stability. A crash at one of the ports/handler leaves the others remain unaffected.
```
Comment (1-Harshit) [inline docs/overview.md:7]: - prefer application
  refer: https://www.geeksforgeeks.org/difference-between-program-and-application/

- Not easy management it is based on microservice arch sorrta neo monolith

File: index.md [hunk 3]
```diff
@@ -3,5 +3,6 @@ title: Home
 layout: home
 nav_order: 1
 ---
-RAS-Backend Documentation
----
\ No newline at end of file
+## RAS Backend Documentation
+RAS is the new recruitment automation system developed by a team of 10 undergraduate students for the Students' Placement Office, IIT Kanpur to ease-out and automate the internship and placement process of the institute.
```
Comment (1-Harshit) [inline index.md:7]: drop new

Comment (1-Harshit) [reply index.md:7]: and what is full form of ras again?

File: docs/errors/errors.md [hunk 4]
```diff
@@ -0,0 +1,9 @@
+---
+title : Logged Errors
+layout : default
+nav_order: 17
+has_children : true
+---
+
+# Common logged errors
+This file lists down the various common errors encountered while developing, building and deploying the application.
```
Comment (1-Harshit) [inline docs/errors/errors.md:9]: <3 Noice file Too many stories i've for this

File: docs/cmd/cmd.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+- RAS
+- Student
+- Company 
+- Admin RC
+- Admin Application 
+- Admin Student
+- Admin Company
+
+---
+#### References
```
Comment (1-Harshit) [inline docs/cmd/cmd.md:27]: nice practice, good

File: docs/cmd/cmd.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
+---
+title : cmd
+layout : default
+nav_order: 7
+has_children: true
+---
+
+# main.go
+
+A mail service is initialized and passed on to the following handlers and the errors are returned to the main function and logged by the logger.
```
Comment (1-Harshit) [inline docs/cmd/cmd.md:10]: initialized in a goroutine, pls note

File: docs/cmd/cmd.md [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+nav_order: 7
+has_children: true
+---
+
+# main.go
+
+A mail service is initialized and passed on to the following handlers and the errors are returned to the main function and logged by the logger. 
+
+{: .info}
+Mail read and write timeouts are defined to 5 seconds and 10 seconds respectively
```
Comment (1-Harshit) [inline docs/cmd/cmd.md:13]: this timeout is of api calls done to the servals endpoints, emailer works almost instantly

File: docs/cmd/cmd.md (hunk 5 above)
Comment (1-Harshit) [inline docs/cmd/cmd.md:27]: nice practice, good

File: docs/cmd/cmd.md [hunk 8]
```diff
@@ -0,0 +1,29 @@
+---
+title : cmd
+layout : default
+nav_order: 7
+has_children: true
+---
+
+# main.go
```
Comment (1-Harshit) [inline docs/cmd/cmd.md:8]: https://github.com/gin-gonic/gin/issues/346

understand and add this somewhere, and put link in refrences

File: docs/cmd/auth.md [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+It returns an http server instance of the started server for reading and writing to the mail service.
+
+The Go viper package is used to read the [config.yaml secrets file]() and get the port of the authentication service. A new Go engine is initialized and the required middlewares are attached to it.
+
+{: .info} 
+The authentication service is hosted on port 3475. 
+
+### Middlewares Attached to the process
+- CORS - Cross-Origin Resource Sharing : Custom middleware defined in the middleware documentation
+- Recovery (Standard Go middleware) : to recover from any panic
```
Comment (1-Harshit) [inline docs/cmd/auth.md:22]: gin middleware

File: docs/cmd/auth.md [hunk 10]
```diff
@@ -0,0 +1,32 @@
+---
+title : auth
+layout : default
+nav_order: 2
+parent : cmd
+---
+
+# auth.go
+
+This file defines mail service handler for the authentication service.
```
Comment (1-Harshit) [inline docs/cmd/auth.md:10]: come again?

File: docs/cmd/auth.md [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+
+## function authServer
+It returns an http server instance of the started server for reading and writing to the mail service.
```
Comment (1-Harshit) [inline docs/cmd/auth.md:13]: why this much focus on mailers?

File: docs/cmd/auth.md [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+
+The Go viper package is used to read the [config.yaml secrets file]() and get the port of the authentication service. A new Go engine is initialized and the required middlewares are attached to it.
```
Comment (1-Harshit) [inline docs/cmd/auth.md:15]: config yml is part of config directory

File: docs/cmd/auth.md [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+
+{: .info} 
+The authentication service is hosted on port 3475. 
+
+### Middlewares Attached to the process
+- CORS - Cross-Origin Resource Sharing : Custom middleware defined in the middleware documentation
+- Recovery (Standard Go middleware) : to recover from any panic
+- Logger (Standard Go Middleware) : To log the status
+
+Initialized service is passed to [auth router].
```
Comment (1-Harshit) [inline docs/cmd/auth.md:25]: and what happens there? put one line of that

File: docs/cmd/auth.md [hunk 14]
```diff
@@ -0,0 +1,32 @@
...
+## function authServer
+It returns an http server instance of the started server for reading and writing to the mail service.
+
+The Go viper package is used to read the [config.yaml secrets file]() and get the port of the authentication service. A new Go engine is initialized and the required middlewares are attached to it.
+
+{: .info} 
+The authentication service is hosted on port 3475. 
+
+### Middlewares Attached to the process
+- CORS - Cross-Origin Resource Sharing : Custom middleware defined in the middleware documentation
```
Comment (1-Harshit) [inline docs/cmd/auth.md:21]: why were these middlewares relevant?

File: docs/cmd/admin.md [hunk 15]
```diff
@@ -0,0 +1,53 @@
+---
+title : admin
+layout : default
+nav_order: 1
+parent : cmd
+---
+
+# admin.go
+This file defines mail service handler for the admin service.
```
Comment (1-Harshit) [inline docs/cmd/admin.md:9]: whaattt

File: docs/cmd/admin.md [hunk 16]
```diff
@@ -0,0 +1,53 @@
...
+
+
+The Go viper package is used to read the [config.yaml secrets file]() and get the port of the various administrator services. A new Go engine is initialized and the required middlewares are attached to it.
```
Comment (1-Harshit) [inline docs/cmd/admin.md:12]: and config.yml is not a secret file, didn't get what u meant here

File: docs/cmd/admin.md [hunk 17]
```diff
@@ -0,0 +1,53 @@
...
+
+
+## function adminRCServer
+It returns an http server instance of the adminRC server for reading and writing to the mail service.
```
Comment (1-Harshit) [inline docs/cmd/admin.md:16]: what is reading and writing to mail service

Comment (1-Harshit) [inline docs/cmd/admin.md:16]: what is reading and writing to mail service
it is a channel for crying out loud

File: docs/cmd/admin.md [hunk 18]
```diff
@@ -0,0 +1,53 @@
...
+
+{: .info} 
+The adminRC service is hosted on port 3490.
+
+### Middlewares Attached to the process
```
Comment (1-Harshit) [inline docs/cmd/admin.md:21]: relevance of middleware?

**Review by 1-Harshit — CHANGES_REQUESTED** [review]

**Comment by 1-Harshit** [comment]
@ahujatejas06 If the PR name isn't decent will not merge
CC: @abhishekshree

File: docs/cmd/admin.md (hunk 15 above)
Comment (abhishekshree) [reply docs/cmd/admin.md:9]: lmao

File: docs/cmd/admin.md (hunk 16 above)
Comment (ahujatejas06) [reply docs/cmd/admin.md:12]: I meant cinfiguration constants... corrected

File: docs/cmd/auth.md (hunk 12 above)
Comment (ahujatejas06) [reply docs/cmd/auth.md:15]: yeah just added a link to that for reference.

**Pull request merged**